rflx/pyrflx/typevalue.py

```diff
@@ -224,39 +237,583 @@ class OpaqueValue(TypeValue):
     def __init__(self, vtype: Opaque) -> None:
         super().__init__(vtype)
 
-    def assign(self, value: bytes, check: bool = True) -> None:
+    def assign(self, value: bytes, length: int = 0, check: bool = True) -> None:
         self._value = value
 
-    def assign_bitvalue(self, value: str, check: bool = True) -> None:
+    def assign_bitvalue(self, value: Bitstring, length: int = 0) -> None:
+        self._value = bytes(value)
 
-        while len(value) % 8 != 0:
-            value = "0" + value
+    @property
+    def length(self) -> int:
+        self._raise_initialized()
+        return len(self._value) * 8
 
-        bytestring = b"".join(
-            [int(value[i : i + 8], 2).to_bytes(1, "big") for i in range(0, len(value), 8)]
-        )
+    @property
+    def value(self) -> bytes:
+        self._raise_initialized()
+        return self._value
+
+    @property
+    def to_bytes(self) -> bytes:
+        self._raise_initialized()
+        return self._value
 
-        self._value = bytestring
+    @property
+    def to_bitstring(self) -> Bitstring:
+        self._raise_initialized()
+        return Bitstring(format(int.from_bytes(self._value, "big"), f"0{self.length}b"))
+
+    @property
+    def accepted_type(self) -> type:
+        return bytes
+
+    @property
+    def literals(self) -> Mapping[Name, Expr]:
+        return {}
+
+
+class ArrayValue(TypeValue):
+
+    _value: List[TypeValue]
+
+    def __init__(self, vtype: Array) -> None:
+        super().__init__(vtype)
+        self._element_type = vtype.element_type
+        self._is_message_array = isinstance(self._element_type, model.Message)
+
+    def assign(self, value: List[TypeValue], length: int = 0, check: bool = True) -> None:
+
+        # check if all elements are of the same class and messages are valid
+        for v in value:
+            if (
+                self._is_message_array
+                and isinstance(v, MessageValue)
+                and isinstance(self._element_type, model.Message)
```

Comment:
If i remove this instance check mypy throws an error because it does not recognize that self._element_type is a Message in this case (although it definitely is).